src/output-tx.c

```diff
@@ -119,12 +120,24 @@ static TmEcode OutputTxLog(ThreadVars *tv, Packet *p, void *thread_data, PacketQ
 
     uint64_t total_txs = AppLayerParserGetTxCnt(p->proto, alproto, alstate);
     uint64_t tx_id = AppLayerParserGetTransactionLogId(f->alparser);
-    int tx_progress_done_value_ts =
-        AppLayerParserGetStateProgressCompletionStatus(p->proto, alproto,
-                                                       STREAM_TOSERVER);
-    int tx_progress_done_value_tc =
-        AppLayerParserGetStateProgressCompletionStatus(p->proto, alproto,
-                                                       STREAM_TOCLIENT);
+
+    int tx_progress_done_value_ts = 0;
+    int tx_progress_done_value_tc = 0;
+
+    /* we want to log TLS once the handshake is done */
+    if (alproto == ALPROTO_TLS) {
+        tx_progress_done_value_ts = TLS_HANDSHAKE_DONE;
```

Comment:
On 27-04-16 11:42, Mats Klepsland wrote:
> In src/output-tx.c
> <https://github.com/inliniac/suricata/pull/2038#discussion_r61230156>:
> 
>> @@ -119,12 +120,24 @@ static TmEcode OutputTxLog(ThreadVars *tv, Packet *p, void *thread_data, PacketQ
>>  
>>      uint64_t total_txs = AppLayerParserGetTxCnt(p->proto, alproto, alstate);
>>      uint64_t tx_id = AppLayerParserGetTransactionLogId(f->alparser);
>> -    int tx_progress_done_value_ts =
>> -        AppLayerParserGetStateProgressCompletionStatus(p->proto, alproto,
>> -                                                       STREAM_TOSERVER);
>> -    int tx_progress_done_value_tc =
>> -        AppLayerParserGetStateProgressCompletionStatus(p->proto, alproto,
>> -                                                       STREAM_TOCLIENT);
>> +
>> +    int tx_progress_done_value_ts = 0;
>> +    int tx_progress_done_value_tc = 0;
>> +
>> +    /* we want to log TLS once the handshake is done */
>> +    if (alproto == ALPROTO_TLS) {
>> +        tx_progress_done_value_ts = TLS_HANDSHAKE_DONE;
> 
> That makes sense :)

I'm thinking that perhaps the cleanest way would be to have another reg
function like 'OutputRegisterTxModule' that either allows you to
register an additional callback for getting the 'ready to log'
'completion' value per direction.

It would likely result in another issue though. If we register 2 loggers
with different callbacks it may mess up the 'log id' tracking.

e.g. have a logger that logs as soon as http_uri is available and one
that logs when tx is complete. This would probably not work in the
suggested scheme.